src/app/codelabs/react/introduction/exercise/test.js

```diff
@@ -0,0 +1,81 @@
+import {
+  babylon,
+  babel_traverse as traverse,
+  babel_types as T
+} from '../code';
+
+export default function test(sourceCode) {
+  const tests = [
+    {
+      instruction: `Create a function named App`,
+      condition: ({ node }) => {
+        const isFunction = T.isFunctionDeclaration(node) && node.id.name === 'App';
+        const isVariable = T.isVariableDeclarator(node) && node.id.name === 'App';
+
+        return isFunction || (isVariable && T.isFunctionExpression(node.init)) || (isVariable && T.ArrowFunctionExpression(node.init))
```

Comment:
You can't use node.id.name in the beginning, because not all nodes have ids, and not all ids are objects. T.isFunctionDeclaration(node) insures that .id.name is accessible